src/vmm/src/device_manager/persist.rs

```diff
@@ -184,6 +229,31 @@ impl<'a> Persist<'a> for MMIODeviceManager {
         let vm = constructor_args.vm;
         let event_manager = constructor_args.event_manager;
 
+        if let Some(balloon_state) = &state.balloon_device {
+            let ctor_args = BalloonConstructorArgs { mem: mem.clone() };
+
+            let device = Arc::new(Mutex::new(
+                Balloon::restore(ctor_args, &balloon_state.device_state).map_err(Error::Balloon)?,
+            ));
+
+            let device_id = balloon_state.device_id.clone();
+            let transport_state = &balloon_state.transport_state;
+            let mmio_slot = &balloon_state.mmio_slot;
```

Comment:
```suggestion
            let mmio_slot = &balloon_state.mmio_slot;
            dev_manager
                .slot_sanity_check(mmio_slot)
                .map_err(Error::DeviceManager)?;
```

Somewhere along the line we should dedup all this code: For every device type code block, after building/restoring the actual device there is the same duplicated code for restoring `MmioTransport` and registering the device to dev-manager and event-manager.